curriculum/js2/9.js

```diff
@@ -4,11 +4,27 @@
  */
 
 const solution = () => {
-  Array.prototype.gsReduce = function () {
-    // To get the actual array, use this
-    return [];
-  };
-};
+  Array.prototype.gsReduce = function (callback, idx = 0, accumulator = 0) {
+    if (typeof idx != "number") {
+      return false;
+    }
+    if (idx == this.length) {
+      return accumulator;
+    }
+    if (typeof accumulator == "number") {
```

Comment:
hmmm can you review reduce function and understand how it works? accumulator does not care what data types it is.